### ISSUE TITLE
[Bug]: Vue3-vite autodocs with docgen 'vue-component-meta' does no longer work with `defineSlots`

### ISSUE THREAD

**Issue by Evertvdw** [storybookjs/storybook#26465]
### Describe the bug

When using the new `vue-component-meta` to generate the docs page of a story it will break when you do that with a component that uses `defineSlots`.  

In the reproduction I created two stories, ButtonSlots and Button. The one which has slots shows these docs:
![image](https://github.com/storybookjs/storybook/assets/5757159/891430b9-350f-42b4-a8bf-c612a2eb7033)
Notice it is missing meta information such as doc comments and default values. It will only show what you are using in the stories or defining in `args`/`argTypes` it seems.

The one without defineSlots looks like this:
![image](https://github.com/storybookjs/storybook/assets/5757159/61b58e01-3ed2-476c-a5f8-baac35c02151)

**Also secondary issue**: HMR will not work for regenerating these autodocs. Only a hard restart of the server will generate updated autodocs, which does not result in a very good DX. If you want to reproduce the issue beware of this :)

### To Reproduce

https://stackblitz.com/edit/github-admhcw?file=src%2Fstories%2FButton2.vue

**Comment by vanessayuenn** [comment]
@larsrickert @chakAs3 is this a known limitation of `vue-component-meta`? Something to be addressed upstream?

**Comment by Evertvdw** [comment]
@vanessayuenn From what I understand volar uses this same package to provide intellisense in VSCode, and I do get proper intellisense on slots when defining them with `defineSlots`. So my guess would be that it is on the storybook side.

**Comment by larsrickert** [comment]
@Evertvdw The issue is that the slot props are not defined. Its a known limitation of `vue-component-meta`, so you need to change it to:

```ts
defineSlots<{
  /**
   * Use for custom content, instead of relying on 'icon' and 'label' props
   */
  default: (props: {}) => VNode[];
  /**
   * Override the default QSpinner when in 'loading' state
   */
  loading: (props: {}) => VNode[];
}>();
```

**Comment by larsrickert** [comment]
I agree to the got reloading tho that would definately be nice to have :)

**Comment by Evertvdw** [comment]
@larsrickert I see thanks! You say it's a known limitation but I could not find this in a quick search inside their repo. Is this documented somewhere? 

Lets hope it will be fixed in the future, in my use case I want to extend some libraries component, and so I would pass an imported interface into defineSlots. If I have to add props I would have to manage that manually :(

**Comment by Evertvdw** [comment]
@larsrickert vue-component-meta@2.0.7 will fix this issue, does this require a version bump on your end?

**Comment by larsrickert** [comment]
@Evertvdw Thanks for the update! Yes I will create a PR for this

**Comment by larsrickert** [comment]
> Any update on this? Cause it still seems to be an issue. 

Which Storybook version are you using? Indon‘t have any issues with the latest 8.2.x

**Comment by valentinpalkovic** [comment]
Hey @Evertvdw, closing the loop on this one. The reported bug is fixed. :slightly_smiling_face:

I reproduced your setup (documented withDefaults props plus defineSlots) against multiple vue-component-meta versions and bisected the fix:

- Root cause: the slot resolver in `vue-component-meta` crashed on slots without parameters (e.g. default?: () => unknown): Cannot read properties of undefined (reading 'flags') on first access of meta.slots. Storybook's docgen plugin swallowed that error and attached no docgen at all, which is why your component lost everything including prop doc comments and defaults, while components without defineSlots were fine.
- Fixed upstream in vue-component-meta 2.0.7 (vuejs/language-tools#4093, released 2024-03-20: nine days after you filed this). Broken in 1.8.x through 2.0.6.
- Current Storybook (@storybook/vue3-vite on 10.x) ships vue-component-meta@^3.2.7, where your exact scenario extracts fully: prop descriptions, default values, and slot doc comments all intact. We added a regression fixture for it to our docgen baseline harness in #35574, so a future dependency bump can't silently reintroduce it.

@karsvaniersel your report (https://github.com/storybookjs/storybook/issues/26465#issuecomment-2251399516) is a different bug, though it landed in this thread: the docs table shows all the union options, but no select control appears. I verified it: the culprit is the null in your union type ('default' | 'destructive' | … | null). The converter in extractArgTypes filters out undefined but not null, and the unquoted null member prevents the literal union from becoming a structured enum type: which is the only shape Controls builds a dropdown from. The description chips come from the type summary, which is why that side looks fine. This is part of the enum/union-controls work currently in flight: #35565 restores dropdowns for literal unions and TS enums, and the null-union case will be handled in the same effort. If it still bites you after that lands, please file a fresh issue and tag me.

Closing as fixed for the original defineSlots report.
